runtime/strategies/search-tokens-to-handles.js

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2017 Google Inc. All rights reserved.
+// This code may only be used under the BSD style license found at
+// http://polymer.github.io/LICENSE.txt
+// Code distributed by Google as part of this project is also
+// subject to an additional IP rights grant found at
+// http://polymer.github.io/PATENTS.txt
+
+import {assert} from '../../platform/assert-web.js';
+import {Strategy} from '../../strategizer/strategizer.js';
+import {Recipe} from '../recipe/recipe.js';
+import {RecipeUtil} from '../recipe/recipe-util.js';
+import {Walker} from '../recipe/walker.js';
+
+export class SearchTokensToHandles extends Strategy {
+  constructor(arc) {
+    super();
+    this._arc = arc;
+  }
+
+  async generate(inputParams) {
+    let arc = this._arc;
+    let findMatchingStores = (token, handle) => {
+      const counts = RecipeUtil.directionCounts(handle);
+      let stores = arc.findHandlesByType(handle.type, {tags: [`#${token}`], subtype: counts.out == 0});
+      let fate = 'use';
+      if (stores.length == 0) {
+        stores = arc._context.findStorageByType(handle.type, {tags: [`#${token}`], subtype: counts.out == 0});
+        fate = counts.out == 0 ? 'map' : 'copy';
+      }
+      stores = stores.filter(store => !handle.recipe.handles.find(handle => handle.id == store.id));
+      return stores.map(store => { return {store, fate, token}; });
+    };
+
+    return Recipe.over(this.getResults(inputParams), new class extends Walker {
+      onHandle(recipe, handle) {
+        // The strategy only works on recipes that have some search tokens resolved to particles/recipes.
```

Comment:
I think we want to start avoiding this kind of implicit in-strategy sequencing. If sequencing is necessary we can use @piotrswigon 's tactics engine.

Reply:
removed the check that search has resolved tokens.